lib/index.js

```diff
@@ -79,35 +77,41 @@ class Client {
 
   /**
    * @param {File} file
-   * @param {string} token
+   * @param {function} onProgress a callback function to track the progress
    */
-  async push(file, token, onProgress) {
+  async _pushBlob(file, onProgress) {
+    // Get the JWT token
+    const token = await this._doBlobAuthz()
+
+    const resource = f11s.open(file)
```

Comment:
I would assume the file object is being *passed* to this method no? Otherwise how will this work in the browser where we don't call `open` but wrap the browser's file object ... (?)

Reply:
See the example code in https://github.com/datopian/ckan-client-js/issues/4

```
// longform of previous item
resource = open(path) # f11s.Resource object
// this uploads just the data to storage ...
// resource has .stream, 
client.pushBlob(resource)
```

Reply:
Also let's use `resource` terminology rather than `file`.

Reply:
@rufuspollock I made a confusing I think  `resource = open(path) # f11s.Resource object`  should be into the SDK and after call `client.pushBlob(resource)`  ?

Question: Do I need to pass `client.pushBlob(resource.stream)`  or `client.pushBlob(resource.buffer)`

Reply:
@mariorodeghiero i just don't understand this comment. Basic usage:

```
// code using the clint
resource = open(path) // or we have a FileInterface object from a browser file object
client.push(resource)
```

> Question: Do I need to pass client.pushBlob(resource.stream) or client.pushBlob(resource.buffer)

What does that mean? Pass to *what*??

@mariorodeghiero please try and be a bit clearer in your questions.

Reply:
ok, Please review the changes and let me know.